File: chat-uikit/src/main/java/io/agora/chat/uikit/chat/interfaces/IChatPrimaryMenu.java
package io.agora.chat.uikit.chat.interfaces;

import android.graphics.drawable.Drawable;
import android.widget.EditText;

import io.agora.chat.uikit.chat.model.EaseInputMenuStyle;


public interface IChatPrimaryMenu {
    /**
     * Set menu display type
     * @param style
     */
    void setMenuShowType(EaseInputMenuStyle style);

    void showNormalStatus();

    void showTextStatus();

    void showVoiceStatus();

    void showEmojiconStatus();

    void showMoreStatus();

    void hideExtendStatus();

    void hideSoftKeyboard();

   /**
     * Enter emoticon event
     * @param emojiContent
     */
    void onEmojiconInputEvent(CharSequence emojiContent);

    /**
     * Delete emoticon event
     */
    void onEmojiconDeleteEvent();

    /**
     * Insert text
     * @param text
     */
    void onTextInsert(CharSequence text);

    /**
     * Get EditText
     * @return
     */
    EditText getEditText();

    /**
     * Set the background of the input box
     * @param bg
     */
    void setMenuBackground(Drawable bg);

    /**
     * Set the send button background
     * @param bg
     */
    void setSendButtonBackground(Drawable bg);

    /**
     * Set up monitoring
     * @param listener
     */
    void setEaseChatPrimaryMenuListener(EaseChatPrimaryMenuListener listener);
}

File: ease-im-kit/src/main/kotlin/io/agora/chat/uikit/feature/chat/interfaces/IChatPrimaryMenu.kt
package io.agora.chat.uikit.feature.chat.interfaces

import android.graphics.drawable.Drawable
import android.widget.EditText
import io.agora.chat.uikit.feature.chat.widgets.ChatUIKitInputMenuStyle

/**
 * The base function of EasePrimaryMenu.
 */
interface IChatPrimaryMenu {
    /**
     * Set menu display type
     * @param style
     */
    fun setMenuShowType(style: ChatUIKitInputMenuStyle?)

    /**
     * Show EditText but hide soft keyboard.
     */
    fun showNormalStatus()

    /**
     * Show EditText and soft keyboard.
     */
    fun showTextStatus()

    /**
     * Show voice style and hide other status.
     */
    fun showVoiceStatus()

    /**
     * Show emoticon extend menu and EditText, hide soft keyboard and other status.
     */
    fun showEmojiconStatus()

    /**
     * Show extend menu and EditText, hide soft keyboard and other status.
     */
    fun showMoreStatus()

    /**
     * Hide extend menu.
     */
    fun hideExtendStatus()

    /**
     * Hide soft keyboard.
     */
    fun hideSoftKeyboard()

    /**
     * Enter emoticon event
     * @param emojiContent
     */
    fun onEmojiconInputEvent(emojiContent: CharSequence?)

    /**
     * Delete emoticon event
     */
    fun onEmojiconDeleteEvent()

    /**
     * Insert text
     * @param text
     */
    fun onTextInsert(text: CharSequence?)

    /**
     * Get EditText
     * @return
     */
    val editText: EditText?

    /**
     * Set the background of the input box
     * @param bg
     */
    fun setMenuBackground(bg: Drawable?)

    /**
     * Set the send button background
     * @param bg
     */
    fun setSendButtonBackground(bg: Drawable?)

    /**
     * Set up monitoring
     * @param listener
     */
    fun setEaseChatPrimaryMenuListener(listener: ChatUIKitPrimaryMenuListener?)

    /**
     * Set the menu visibility
     */
    fun setVisible(visible: Int)
}